helix-term/src/ui/picker.rs

```diff
@@ -403,13 +539,22 @@ impl<T: Item + 'static> Picker<T> {
     ) -> Preview<'picker, 'editor> {
         match path_or_id {
             PathOrId::Path(path) => {
-                let path = &path;
+                let path_arc = Arc::new(path);
```

Comment:
You are still copying the path buff here. To avoid that you would need to change the type in `PathOrId` (and the preview hashmap).

You may even be able to get away with just changing it in the hashmap and using `entry(path).unwarp().key().clone()`. It's a bit hacks but would require less changes. I wouldn't bleed the `Arc<Path>` out to public API but I think PathORId is private IIRC so changing it should be fine and would feel a bit cleaner

Reply:
I moved `PathOrId::Path` to use an `Arc<Path>` since the `From` implementation makes it really easy to take advantage of and `PathOrId` is nicely private to the picker module :+1: